### PR TITLE
CMake: migrates changes for Levante to AWIESM 2.1 branch

### PR DESCRIPTION
**Pull request by pgierz**
Includes changes to the build environment discussed in https://github.com/FESOM/fesom2/discussions/336 for the AWI-ESM version of FESOM 2.1. I would please ask @hegish to check if this makes sense in the CMake world and @mandresm or @JanStreffing if it mirrors what you needed to do for AWI-CM3

### CODE REVIEW

**Comment by pgierz** [comment]
This PR also includes changes to make the tests run correctly. The failure in the testing suite is an in-complete separation of the iceberg code, and should probably be looked at by @ackerlar.

File: env/mistral.dkrz.de/shell-intel+openmpi [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+export GRIBROOT=/pf/a/a270092/ecmwf/grib_api_intel_modulegcc
+export NETCDFFROOT=/sw/rhel6-x64/netcdf/netcdf_fortran-4.4.3-intel14
+export NETCDFROOT=/sw/rhel6-x64/netcdf/netcdf_c-4.4.0-gcc48
+export HDF5ROOT=/sw/rhel6-x64/hdf5/hdf5-1.8.14-threadsafe-gcc48
+export SZIPROOT=/sw/rhel6-x64/sys/libaec-0.3.2-gcc48
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:${NETCDFFROOT}/lib:${HDF5ROOT}/lib:${NETCDFROOT}/lib:${SZIPROOT}/lib:${GRIBROOT}/lib
+
+export FESOM_USE_CPLNG='active'
+
+export DR_HOOK=1
```
Comment (JanStreffing) [inline env/mistral.dkrz.de/shell-intel+openmpi:45]: Are we using DR_HOOK in FESOM2 now? DR_HOOK is the IFS internal debug program. See: https://confluence.ecmwf.int/download/attachments/19661682/drhook.pdf?version=1&modificationDate=1443002579511&api=v2

If we don't use it, we should not set any flag for it here.

**Comment by hegish** [comment]
Hey @pgierz, I did not spot anything while skimming through. But there are many changes. Anything special I should look out for?

**Comment by pgierz** [comment]
Hi @hegish, no, just a quick check if none of the CMakeLists look as if they are out of place. So, if you give a thumbs up, I will assume nothing is fishy.

In that case, there are still two changes needed, one @JanStreffing made me aware of, which I will take care after my holiday, and then @ackerlar still needs to put proper switches around his iceberg code. Currently, if the iceberg code is turned off, FESOM will not compile. Provisionally, I will just hard-wire it to be turned on.

**Review by mandresm — APPROVED** [review]

**Pull request merged**